Source/WTF/wtf/MathExtras.h

```diff
@@ -499,9 +499,9 @@ inline typename std::enable_if<std::is_floating_point<T>::value, T>::type nanPro
     return std::isnan(a) || std::isnan(b) ? std::numeric_limits<T>::quiet_NaN() : std::max(a, b);
 }
 
-inline bool isIntegral(float value)
+inline constexpr bool isIntegral(float value)
 {
-    return static_cast<int>(value) == value;
+    return std::trunc(value) == value;
```

Comment:
This changes behavior and, I believe, regresses correctness for _infinity_ and _nan_; the function now returns _true_ and before this change it returned _false_. Maybe that’s OK given how we use the function.

Reply:
I think both implementations appear to return `false` when given `NaN`. However, as noted, the behavior with `infinity` has changed.
In certain environments, specific values such as `std::numeric_limits<float>::min()` may lead to runtime errors or produce incorrect results with the previous logic. Given that, I believe this change is justified.

Added test cases in #45264 to ensure the expected `isIntegral` behavior :)

Reply:
Thanks for highlighting this!